index.ts

```diff
@@ -192,34 +193,45 @@ interface Card {
   name: string;
 }
 
-let rooms: Room[] = [];
+let joinRooms: {[roomId: string]: Room} = {};
+let viewRooms: {[roomId: string]: Room} = {};
 wss.on("request", async function (request) {
+  const roomId = request.resourceURL.path.substr(1);
+  const readOnly = !!viewRooms[roomId];
```

Comment:
This breaks when opening `/stimmung/<valid sha256>`: the client tries to connect as participant but the server will detect it here as `readOnly`.
IMHO `readOnly` should be specified explicitly by the client when `join`ing. That way we known which UI is used by the client and collisions between `roomId`s and `viewId`s are not relevant anymore.